### ISSUE TITLE
page.get_text(clip=rect) returns empty string in some cases

### ISSUE THREAD

**Issue by Don-Yin** [pymupdf/PyMuPDF#1741]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
Hi Jorj, 

I discovered ```page.get_text(clip=rect)``` returns empty string when the ```rect``` is small. Is this to be expected?

I'm attempting to extract highlighted text using code from this [this discussion](https://github.com/pymupdf/PyMuPDF/discussions/1671)  and your code from [this issue](https://github.com/pymupdf/PyMuPDF/issues/318), and there appear to be two ways to do so:

[Method 1](https://github.com/pymupdf/PyMuPDF/issues/318) (suggested by you):
```python
for annot in page.annots(types=(fitz.PDF_ANNOT_HIGHLIGHT, fitz.PDF_ANNOT_UNDERLINE)):
    print(page.get_text(clip=annot.rect))
```

[Method 2](https://github.com/pymupdf/PyMuPDF/issues/318):
```python
def _extract_annot(annot, words_on_page):
    """Extract words in a given highlight.
    Args:
        annot: fitz.Annot
        words_on_page (list): [description]
    Returns:
        str: words in the entire highlight.
    """
    quad_count = int(len(annot.vertices) / 4)
    sentences = ["" for i in range(quad_count)]
    for i in range(quad_count):
        points = annot.vertices[i * 4 : i * 4 + 4]
        words = [w for w in words_on_page if _check_contain(fitz.Rect(w[:4]), points)]
        sentences[i] = " ".join(w[4] for w in words)
    sentence = " ".join(sentences)
    return sentence
```

Both approaches have drawbacks:
Method 1: it appears that this method treats the entire highlight rect as a giant square (the entire image below), and thus returns text outside of the actual highlight area (indicated by the blue outline):

<img width="463" alt="Screenshot 2022-06-02 at 08 13 59" src="https://user-images.githubusercontent.com/65135356/171574469-e1c8f21b-49c7-46e4-bfa0-aa6faf54f367.png">

Undesired text: "ton for higher order thinking." + "Specifically, analogical reasoning, long"

Method 2: it works, but it feels hacky and takes long.

So I thought I could combine these methods by using your ```page.get_text(clip=annot.rect)``` instead of ```words = [w for w in words_on_page if _check_contain(fitz.Rect(w[:4]), points)]```. Essentially what this does is breaking down the giant highlight rect into small ones and perform ```page.get_text(clip=annot.rect)``` on each individual rects. I printed them out using ```pixmap```, they look like this:

![(48 0, 268 02001953125), (289 5, 268 02001953125), (48 0, 279 0899963378906), (289 5, 279 0899963378906)](https://user-images.githubusercontent.com/65135356/171576202-88a2c7a1-a7f6-4e42-a674-f4e27178ffb0.png)
![(48 0, 280 0199890136719), (289 4800109863281, 280 0199890136719), (48 0, 291 0899963378906), (289 4800109863281, 291 0899963378906)](https://user-images.githubusercontent.com/65135356/171576207-65016427-992b-4eaf-9dc9-facc1c328461.png)
![(48 0, 292 0199890136719), (127 76000213623047, 292 0199890136719), (48 0, 303 0899963378906), (127 76000213623047, 303 0899963378906)](https://user-images.githubusercontent.com/65135356/171576212-831cc5b2-f9ae-4749-a3fb-77e2eaf93fdd.png)
![(175 30999755859375, 256 02001953125), (289 4700012207031, 256 02001953125), (175 30999755859375, 267 09002685546875), (289 4700012207031, 267 09002685546875)](https://user-images.githubusercontent.com/65135356/171576215-a06cf5a5-7f54-4c92-b964-23df904558cb.png)

So what I tried is this:
```python
def _extract_annot(annot, page):
    """Extract words in a given highlight.
    Args:
        annot: fitz.Annot
        words_on_page (list): [description]
    Returns:
        str: words in the entire highlight.
    """
    quad_count = int(len(annot.vertices) / 4)
    sentences = ["" for i in range(quad_count)]
    for i in range(quad_count):
        points = annot.vertices[i * 4 : i * 4 + 4]
        sentences[i] = page.get_text(clip=fitz.Quad(points).rect)
    sentences = " ".join(sentences)
    return sentences
```

However, this simply returns some empty strings.

Is this typical behaviour of ``` page.get_text(clip=rect)```? Maybe not a bug? Is there anything I can do to make it work?

## To Reproduce (mandatory)
Explain the steps to reproduce the behavior, For example, include a minimal code snippet, example files, etc.
The example pdf: 
[paper1.2016_unknown_010517_297287.pdf](https://github.com/pymupdf/PyMuPDF/files/8821669/paper1.2016_unknown_010517_297287.pdf)
The highlight is on page one.
Code:
```python
def _extract_annot(annot, page):
    """Extract words in a given highlight.
    Args:
        annot: fitz.Annot
        words_on_page (list): [description]
    Returns:
        str: words in the entire highlight.
    """
    quad_count = int(len(annot.vertices) / 4)
    sentences = ["" for i in range(quad_count)]
    for i in range(quad_count):
        points = annot.vertices[i * 4 : i * 4 + 4]
        sentences[i] = page.get_text(clip=fitz.Quad(points).rect)
    sentences = " ".join(sentences)
    return sentences

doc = fitz.open("paper1.2016_unknown_010517_297287.pdf")

page = doc[1]

for annot in page.annots():
    text = _extract_annot(annot, page)
    print(text)
```

## Your configuration (mandatory)
 - Operating system, potentially version and bitness: MacOs
 - Python version, bitness: 3.10.4
 - PyMuPDF version, installation method (**wheel** or **generated** from source): 1.19.6 using pip

Best,
Don

**Comment by JorjMcKie** [comment]
> I discovered page.get_text(clip=rect) returns empty string when the rect is small. Is this to be expected?

Of course. **_This is not a bug._** If you miss even the smallest part of a character, it will not be included.
So either use `fitz.TOOLS.set_small_glyph_heights(True)` to at least cut off foot and head room of each character, or enlarge the clip rectangles a bit (e.g. by 1 in every direction) and use something like `clip=fitz.Quad(points).rect + (-1, -1, 1, 1)`.
